docs/Trainer/Checkpointing.md

```diff
@@ -26,11 +27,26 @@ checkpoint_callback = ModelCheckpoint(
 trainer = Trainer(checkpoint_callback=checkpoint_callback)
 ```
 
+The `save_top_k` options works in the following ways:
+
+| save_top_k        | behavior |
+| --------   | -----  |
+| 0     | no models are saved |
+| -1        |  all models are saved  |
+|  k >= 1        |  the best k models are saved  |
+
+
+Also, if `save_top_k` >= 2 and the callback is called multiple
+times inside an epoch, the name of the saved file will be
+appended with a version count starting with `v0`.
```

Comment:
I understand this PR has been out for a while so it's okay if this change isn't implemented, but at latent space we've implemented a checkpointing system within an epoch (we have a usecase of epochless datasets) and we rely heavily on having the iteration in the name for some of our checkpoint analysis. If we don't think we're saving multiple times in the same iteration, we could consider using the iteration number

Reply:
+1 on considering iteration numbers.
But I think it's too much for this PR, perhaps considering a seperate issue?
You see we expect so many things from this single PR, and it is blocked like forever. And I have to merge master from time to time.